google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/SubscriberImpl.java

```diff
@@ -96,62 +111,65 @@ public SubscriberImpl(SubscriberImpl.Builder builder) throws IOException {
                 .sslContext(GrpcSslContexts.forClient().ciphers(null).build())
                 .executor(executor);
 
-    Credentials credentials =
+    credentials =
         builder.credentials.isPresent()
             ? builder.credentials.get()
             : GoogleCredentials.getApplicationDefault()
                 .createScoped(Collections.singletonList(PUBSUB_API_SCOPE));
 
-    FlowController flowController =
-        new FlowController(builder.maxOutstandingBytes, builder.maxOutstandingBytes, false);
-    subscriberConnections = new SubscriberConnection[numChannels];
-    for (int i = 0; i < subscriberConnections.length; i++) {
-      subscriberConnections[i] =
-          new SubscriberConnection(
-              subscription,
-              credentials,
-              builder.receiver,
-              ackExpirationPadding,
-              streamAckDeadlineSeconds,
-              ackLatencyDistribution,
-              channelBuilder.build(),
-              flowController,
-              executor,
-              clock);
-    }
+    streamingSubscriberConnections = new ArrayList<StreamingSubscriberConnection>(numChannels);
+    pollingSubscriberConnections = new ArrayList<PollingSubscriberConnection>(numChannels);
```

Comment:
I actually argue it will make the code less error prone and also requires adding a clunky casting